### PR TITLE
Remove builtin exception types from protobuf and ASTs.

### PR DESCRIPTION
**Pull request by sofiafaro-da**
Removes BTGeneralError, BTArithmeticError, BTContractError and associated primitives.
Adds BEAnyExceptionIsArithmeticError and BEAnyExceptionIsContractError.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala [hunk 1]
```diff
@@ -588,18 +588,13 @@ private[lf] final class Compiler(
...
-          case BGeneralErrorMessage | BArithmeticErrorMessage | BContractErrorMessage =>
-            SBBuiltinErrorMessage
-
           case BAnyExceptionMessage =>
             SBAnyExceptionMessage
+          case BAnyExceptionIsArithmeticError =>
+            throw CompilationError("SBAnyExceptionIsArithmeticError not implemented")
+          case BAnyExceptionIsContractError =>
+            throw CompilationError("SBAnyExceptionIsContractError not implemented")
+
```
Comment (cocreature) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:597]: Please add TODOs

File: daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/TypingSpec.scala [hunk 2]
```diff
@@ -65,9 +65,6 @@ class TypingSpec extends AnyWordSpec with TableDrivenPropertyChecks with Matcher
         BTRoundingMode -> k"*",
```
Comment (cocreature) [inline daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/TypingSpec.scala:65]: Can we add some tests for the new primitives?

**Review by cocreature — APPROVED** [review]
Nice, thank you!

File: daml-lf/interface/src/main/scala/com/digitalasset/daml/lf/iface/reader/InterfaceReader.scala [hunk 3]
```diff
@@ -246,8 +246,7 @@ object InterfaceReader {
         case Ast.BTOptional => \/-((1, PrimType.Optional))
         case Ast.BTTextMap => \/-((1, PrimType.TextMap))
         case Ast.BTGenMap => \/-((2, PrimType.GenMap))
-        case Ast.BTAnyException | Ast.BTGeneralError | Ast.BTArithmeticError | Ast.BTContractError |
-            Ast.BTBigNumeric | Ast.BTRoundingMode =>
+        case Ast.BTAnyException | Ast.BTBigNumeric | Ast.BTRoundingMode =>
           // TODO https://github.com/digital-asset/daml/issues/8020
           // Add exception types to the interface reader
```
Comment (remyhaemmerle-da) [inline daml-lf/interface/src/main/scala/com/digitalasset/daml/lf/iface/reader/InterfaceReader.scala:251]:
```suggestion
```

File: daml-lf/interface/src/test/scala/com/digitalasset/daml/lf/iface/TypeSpec.scala [hunk 4]
```diff
@@ -80,8 +80,7 @@ class TypeSpec extends AnyWordSpec with Matchers {
           case Pkg.BTTypeRep => sys.error("cannot use type representation in interface type")
           case Pkg.BTRoundingMode => sys.error("cannot use rounding mode in interface type")
           case Pkg.BTBigNumeric => sys.error("cannot use big numeric in interface type")
-          case Pkg.BTAnyException | Pkg.BTArithmeticError | Pkg.BTContractError |
-              Pkg.BTGeneralError =>
+          case Pkg.BTAnyException =>
             // TODO https://github.com/digital-asset/daml/issues/8020
```
Comment (remyhaemmerle-da) [inline daml-lf/interface/src/test/scala/com/digitalasset/daml/lf/iface/TypeSpec.scala:84]:
```suggestion
```

**Review by remyhaemmerle-da — APPROVED** [review]
LGTM

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala (hunk 1 above)
Comment (sofiafaro-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala]: Added!

File: daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/TypingSpec.scala (hunk 2 above)
Comment (sofiafaro-da) [reply daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/TypingSpec.scala:65]: Added!

**Comment by remyhaemmerle-da** [comment]
part of #8020

**Comment by remyhaemmerle-da** [comment]
part of #8020

**Pull request merged**